src/main/java/com/microsoft/sqlserver/jdbc/ActivityCorrelator.java

```diff
@@ -33,7 +31,7 @@ static ActivityId getCurrent() {
 
         // Since the Id for each thread is unique, this assures that the below if statement is run only once per thread.
         if (!activityIdTlsMap.containsKey(uniqueThreadId)) {
-            activityIdTlsMap.put(uniqueThreadId, new ActivityId());
+            activityIdTlsMap.put(uniqueThreadId, new ActivityId(uniqueThreadId));

```

Comment:
Anywhere that calls getCurrent() is going to end up putting an entry in the map in the context of the current thread. In a pooled environment, each thread that uses the connection could potentially do that and then only the pooling thread that closes the connection will remove the original thread's entry from the map. There will still be a leak.

Reply:
Thanks. I'm looking into the documentation as to if Activity IDs are tied to a connection (one ActivityID per connection) or if it's tied to other queries as well. Depending on that, I think we can come up with a solution that doesn't affect the driver's performance - I'll come up with an update soon.